[0.21] test: replace hashlib.ripemd160 with an own implementation

fanquake opened the pull request:
Backports #23716 to 0.21. Closes #25534.

sipa committed:
Add pure Python RIPEMD-160

Github-Pull: #23716
Rebased-From: ad3e9e1f214d739e098c6ebbd300da5df1026a44

sipa committed:
Swap out hashlib.ripemd160 for own implementation

Github-Pull: #23716
Rebased-From: 5b559dc7ecf37ab1604b75ec8ffe8436377a5fb1

MarcoFalke commented:
Funny that this runs a travis build, even though we deleted travis from this repo: https://travis-ci.org/github/bitcoin/bitcoin/builds/774819450

fanquake commented:
> Funny that this runs a travis build, even though we deleted travis from this repo: 

There was a Travis webhook left in https://github.com/bitcoin/bitcoin/settings/hooks. I have removed that now.

MarcoFalke commented:
Did the cherry-pick myself and it matched
